control-plane/pkg/reconciler/sink/controller.go

```diff
@@ -29,21 +29,24 @@ import (
 	secretinformer "knative.dev/pkg/client/injection/kube/informers/core/v1/secret"
 	"knative.dev/pkg/configmap"
 	"knative.dev/pkg/controller"
-	"knative.dev/pkg/logging"
+	pkglogging "knative.dev/pkg/logging"
 	"knative.dev/pkg/tracker"
 
 	eventing "knative.dev/eventing-kafka-broker/control-plane/pkg/apis/eventing/v1alpha1"
 	sinkinformer "knative.dev/eventing-kafka-broker/control-plane/pkg/client/injection/informers/eventing/v1alpha1/kafkasink"
 	sinkreconciler "knative.dev/eventing-kafka-broker/control-plane/pkg/client/injection/reconciler/eventing/v1alpha1/kafkasink"
 	"knative.dev/eventing-kafka-broker/control-plane/pkg/config"
+	"knative.dev/eventing-kafka-broker/control-plane/pkg/logging"
 	"knative.dev/eventing-kafka-broker/control-plane/pkg/reconciler/base"
 )
 
 func NewController(ctx context.Context, _ configmap.Watcher, configs *config.Env) *controller.Impl {
 
 	eventing.RegisterConditionSet(base.IngressConditionSet)
 
-	logger := logging.FromContext(ctx)
+	logger := pkglogging.FromContext(ctx)
+
+	sarama.Logger = logging.NewSaramaLogger(logger)
```

Comment:
Initializing Sarama's logger right here, in the middle of a `NewController` constructor, is absolutely disgusting.

Ideally I would do it in the control-plane's `main`, but at the point Knative hasn't initialized its logger yet.

The only alternative I can think of is to initialize our own zap logger in the `main`, with the limitation that this logger main not inherit from Knative's dynamic configmap-based configuration.

Reply:
(side note: We should probably decorate the logger with a new name, like `sarama` instead of the default `kafka-broker-controller`)

Reply:
Yeah, especially since this is a global configuration so it will impact other controllers too.

> The only alternative I can think of is to initialize our own zap logger in the main, with the limitation that this logger may not inherit from Knative's dynamic configmap-based configuration.

I can't think of another alternative too, this should be good enough for now.